src/main/kotlin/com/sourcegraph/cody/ignore/IgnoreOracle.kt

```diff
@@ -85,4 +85,18 @@ class IgnoreOracle(private val project: Project) {
     }
     return completable
   }
+
+  /**
+   * Gets whether `uri` should be ignored for autocomplete, etc. If the result is not available
+   * quickly, returns null and invokes `orElse` on a pooled thread when the result is available.
+   */
+  fun policyForUriOrElse(uri: String, orElse: (policy: IgnorePolicy) -> Unit): IgnorePolicy? {
+    val completable = policyForUri(uri)
+    try {
+      return completable.get(16, TimeUnit.MILLISECONDS)
```

Comment:
So in fact we want to either get it from cache or get a timeout?

16 milliseconds looks weird to me, It think I would be cleaner this way:

```kt
val result = policyForUri(uri).getNow(null)
if (result == null) ApplicationManager.getApplication().executeOnPooledThread { orElse(completable.get()) }
return result
```

Reply:
My thinking with the timeout was to retrieve it if Agent is up and responding quickly. Otherwise, we will be refreshing editor banners every time you open a file, because the first retrieval will always miss the cache.